src/BaseFirestoreRepository.ts

```diff
@@ -102,8 +102,9 @@ export class BaseFirestoreRepository<T extends IEntity> extends AbstractFirestor
     queries: Array<IFireOrmQueryLine>,
     limitVal?: number,
     orderByObj?: IOrderByParams,
-    single?: boolean
-  ): Promise<T[]> {
+    single?: boolean,
+    onUpdate?: (documents: T[]) => void
+  ): Promise<any> {
```

Comment:
Given than `onUpdate` returns nothing, can we update the signature of this method to be: `Promise<T[]> || () => void`?

Reply:
Sorry I'm not sure I understand, but there is a definite issue with this:

`query.onSnapshot` returns `() => void` so the `Promise<any>` return type of the `execute` function is definitely misleading. Do you think I should wrap `query.onSnapshot` in a promise?

I can also change the return type of `execute` to `Promise<T[] | (() => void)>` which works but is also misleading because `query.onSnapshot` doesn't currently return a promise.

Reply:
Actually, I think I am wrong about this ^ I'll think about how to make it more clear but let me know if you have any notes.

Reply:
I attempted to make all the types promises, let me know what you think https://github.com/garviand/fireorm/pull/2/files